Gem/Code/Source/Components/Multiplayer/GameplayEffectsComponent.cpp

```diff
@@ -0,0 +1,160 @@
+
+        m_soundTriggerNames[aznumeric_cast<int>(SoundEffect::LaserPistolMuzzleFlash)] = GetLaserPistolMuzzleFlash();
+        m_soundTriggerNames[aznumeric_cast<int>(SoundEffect::LaserPistolImpact)] = GetLaserPistolImpact();
+
+        m_soundTriggerNames[aznumeric_cast<int>(SoundEffect::BubbleGunBuildup)] = GetBubbleGunBuildup();
+        m_soundTriggerNames[aznumeric_cast<int>(SoundEffect::BubbleGunMuzzleFlash)] = GetBubbleGunMuzzleFlash();
+        m_soundTriggerNames[aznumeric_cast<int>(SoundEffect::BubbleGunProjectile)] = GetBubbleGunProjectile();
+        m_soundTriggerNames[aznumeric_cast<int>(SoundEffect::BubbleGunImpact)] = GetBubbleGunImpact();
+
+        m_soundTriggerNames[aznumeric_cast<int>(SoundEffect::JumpPadLaunch)] = GetJumpPadLaunch();
+
+        m_soundTriggerNames[aznumeric_cast<int>(SoundEffect::EnergyBallTrapRisingOutOfTheGround)] = GetEnergyBallTrapRisingOutOfTheGround();
+        m_soundTriggerNames[aznumeric_cast<int>(SoundEffect::EnergyBallTrapBuildup)] = GetEnergyBallTrapBuildup();
+        m_soundTriggerNames[aznumeric_cast<int>(SoundEffect::EnergyBallTrapProjectile)] = GetEnergyBallTrapProjectile();
+        m_soundTriggerNames[aznumeric_cast<int>(SoundEffect::EnergyBallTrapImpact)] = GetEnergyBallTrapImpact();
+        m_soundTriggerNames[aznumeric_cast<int>(SoundEffect::EnergyBallTrapOnCooldown)] = GetEnergyBallTrapOnCooldown();
+    }
+
+    void GameplayEffectsComponent::OnDeactivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+        Audio::AudioTriggerNotificationBus::MultiHandler::BusDisconnect();
+    }
+
+    void GameplayEffectsComponent::HandleRPC_OnEffect([[maybe_unused]] AzNetworking::IConnection* invokingConnection,
+        const SoundEffect& effect)
+    {
+        // Spawn at the local camera position
+        AZ::EntityId camera;
+        Camera::CameraSystemRequestBus::BroadcastResult(camera, &Camera::CameraSystemRequestBus::Events::GetActiveCamera);
+        if (camera.IsValid())
+        {
+            AZ::Vector3 cameraPosition = AZ::Vector3::CreateZero();
+            AZ::TransformBus::EventResult(cameraPosition, camera, &AZ::TransformBus::Events::GetWorldTranslation);
+            SpawnEffect(effect, cameraPosition);
+        }
+
+    }
+
+    void GameplayEffectsComponent::HandleRPC_OnPositionalEffect([[maybe_unused]] AzNetworking::IConnection* invokingConnection,
+        const SoundEffect& effect, const AZ::Vector3& soundLocation)
+    {
+        SpawnEffect(effect, soundLocation);
+    }
+
+    void GameplayEffectsComponent::ReportTriggerFinished([[maybe_unused]] Audio::TAudioControlID triggerId)
+    {
+        const Audio::TriggerNotificationIdType busId = *Audio::AudioTriggerNotificationBus::GetCurrentBusId();
+        const AZ::EntityId entityId = aznumeric_cast<AZ::EntityId>(busId.m_owner);
+        Audio::AudioTriggerNotificationBus::MultiHandler::BusDisconnect(busId);
```

Comment:
Here, if an Entity might have multiple concurrent sounds playing, disconnecting in this manner would disconnect from all Finished notifications.  You could come up with a way to keep the connection if there are other spawned effects, or something like that.  Or keeping the bus connected for longer could be ok too, disconnect all during Deactivate.

Reply:
Understood. As it is designed right now, there is only one sound effect per spawned entity. So this should be fine.